pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -279,6 +286,14 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 			return result, nil
 		}
 
+		if host.HasError() {
+			// We have tried to register and validate the host and
+			// that failed in a way we assume is not retryable, so do
+			// not proceed to any other steps.
+			reqLogger.Info("registration error")
```

Comment:
Do we assume that the error information in the CR status gives the user some direction about what they should do to fix the situation?

And is it safe to assume that fixing it will always involve some API action that causes reconciliation to run again? I just want to make sure there is some path to a user taking action to fix the problem and then re-reconciling.